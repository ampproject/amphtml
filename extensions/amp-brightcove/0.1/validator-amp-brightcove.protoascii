#
# Copyright 2016 The AMP HTML Authors. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS-IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the license.
#

tags: {  # amp-brightcove
  html_format: AMP
  html_format: AMP4ADS
  tag_name: "SCRIPT"
  spec_name: "amp-brightcove extension .js script"
  mandatory_parent: "HEAD"
  unique_warning: true
  extension_unused_unless_tag_present: "amp-brightcove"
  attrs: {
    name: "async"
    mandatory: true
    value: ""
  }
  attrs: {
    name: "custom-element"
    mandatory: true
    value: "amp-brightcove"
    dispatch_key: true
  }
  attrs: { name: "nonce" }
  attrs: {
    name: "src"
    mandatory: true
    value_regex: "https://cdn\\.ampproject\\.org/v0/amp-brightcove-(latest|0\\.1).js"
  }
  attrs: {
    name: "type"
    value_casei: "text/javascript"
  }
  cdata: {
    blacklisted_cdata_regex: {
      regex: "."
      error_message: "contents"
    }
  }
  spec_url: "https://www.ampproject.org/docs/reference/extended/amp-brightcove.html"
}
tags: {  # <amp-brightcove>
  html_format: AMP
  html_format: AMP4ADS
  tag_name: "AMP-BRIGHTCOVE"
  disallowed_ancestor: "AMP-SIDEBAR"
  also_requires_tag: "amp-brightcove extension .js script"
  attrs: {
    name: "data-account"
    mandatory: true
  }
  # If data-embed is missing, the default value is "default".
  attrs: { name: "data-embed" }
  # If data-player is missing, the default value is "default".
  attrs: { name: "data-player" }
  # If data-playlist-id is not used for perform player and hence it is optional.
  attrs: { name: "data-playlist-id" }
  # If data-video-id is not used for perform player and hence it is optional.
  attrs: { name: "data-video-id" }
  attr_lists: "extended-amp-global"
  spec_url: "https://www.ampproject.org/docs/reference/extended/amp-brightcove.html"
  amp_layout: {
    supported_layouts: FILL
    supported_layouts: FIXED
    supported_layouts: FIXED_HEIGHT
    supported_layouts: FLEX_ITEM
    supported_layouts: NODISPLAY
    supported_layouts: RESPONSIVE
  }
}
