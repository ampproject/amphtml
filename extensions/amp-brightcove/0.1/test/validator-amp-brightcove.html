<!--
  Copyright 2015 The AMP HTML Authors. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS-IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the license.
-->
<!--
  Test Description:
  Tests support for the amp-brightcove tag.
-->
<!doctype html>
<html âš¡>
<head>
  <meta charset="utf-8">
  <link rel="canonical" href="./regular-html-version.html">
  <meta name="viewport" content="width=device-width,minimum-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-brightcove" src="https://cdn.ampproject.org/v0/amp-brightcove-0.1.js"></script>
</head>
<body>
  <!-- Valid: Example of a valid amp-brightcove. -->
  <amp-brightcove
      autoplay
      data-account="1290862519001"
      data-video-id="ref:amp-docs-sample"
      data-player-id="SyIOV8yWM"
      layout="responsive" width="480" height="270">
  </amp-brightcove>
  <!-- Valid: data-video-id, data-player-id are optional;
       leaving them out results in the default config. -->
  <amp-brightcove
      data-account="1290862519001"
      layout="responsive" width="480" height="270">
  </amp-brightcove>
  <!-- Valid: most "data-*" attributes are bindable. -->
  <amp-brightcove
      data-account="1290862519001"
      layout="responsive" width="480" height="270"
      [data-account]="foo.bar" [data-embed]="foo.bar" [data-player]="foo.bar"
      [data-player-id]="foo.bar" [data-playlist-id]="foo.bar"
      [data-video-id]="foo.bar">
  </amp-brightcove>
  <!-- Invalid: data-account is mandatory;
       leaving them out results in an error. -->
  <amp-brightcove
      layout="responsive" width="480" height="270">
  </amp-brightcove>
</body>
</html>
