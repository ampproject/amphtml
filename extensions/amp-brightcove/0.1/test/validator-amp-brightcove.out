FAIL
amp-brightcove/0.1/test/validator-amp-brightcove.html:54:2 The mandatory attribute 'data-account' is missing in tag 'amp-brightcove'. (see https://www.ampproject.org/docs/reference/components/amp-brightcove) [AMP_TAG_PROBLEM]
